### DISCUSSION TITLE
Run the server with asyncio.run(...)

### DISCUSSION THREAD

**Discussion by szabolcsdombi** [aio-libs/aiohttp#6517] [Q&A] [answered]
### Is your feature request related to a problem?

I cannot run other async tasks alongside the server.
I managed to run `websockets.serve(...)` and a background task running with:

```py
async def main():
    await asyncio.gather(websocket_server(), background_task())

asyncio.run(main())
```

Is this possible with **aiohttp**?

### Describe the solution you'd like

I would like to run the aiohttp server alongside other async tasks similar to the example above.

### Describe alternatives you've considered

Found `aiojobs`, but what I want to achieve is to run `aiohttp` within an existing project with `asyncio`.

### Related component

Server

### Code of Conduct

- [X] I agree to follow the aio-libs Code of Conduct

**Comment by asvetlov** [comment, marked as answer]
You can use https://docs.aiohttp.org/en/stable/web_advanced.html#aiohttp-web-app-runners for starting the application without `run_app` call.

**Reply by Dreamsorcerer** [reply]
Generally, it's preferable to run other tasks spawned from the aiohttp app and allow aiohttp to control the loop: https://docs.aiohttp.org/en/stable/web_advanced.html#complex-applications

But, if that's not feasible, then use the app runners as mentioned above and manage the loop yourself.

**Comment by szabolcsdombi** [comment]
Thank you, the code from [aiohttp-web-app-runners](https://docs.aiohttp.org/en/stable/web_advanced.html#aiohttp-web-app-runners) worked well.
